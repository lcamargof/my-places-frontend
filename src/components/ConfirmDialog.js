import React from 'react';
import PropTypes from 'prop-types';
import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from 'material-ui';

const ConfirmDialog = ({ place, open, onClose, onConfirm }) => (
  <Dialog
    open={open}
    onClose={onClose}
    aria-labelledby="alert-dialog-title"
    aria-describedby="alert-dialog-description"
  >
    <DialogTitle id="alert-dialog-title"> Remove a place :(</DialogTitle>
    <DialogContent>
      <DialogContentText id="alert-dialog-description">
        Are you sure you want to remove "{place.name}" from your list of places?
      </DialogContentText>
    </DialogContent>
    <DialogActions>
      <Button onClick={onClose}>
        Cancel
      </Button>
      <Button onClick={onConfirm} color="primary" autoFocus>
        DELETE
      </Button>
    </DialogActions>
  </Dialog>
);

ConfirmDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  place: PropTypes.object,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func.isRequired,
};

ConfirmDialog.defaultProps = {
  place: {},
};

export default ConfirmDialog;