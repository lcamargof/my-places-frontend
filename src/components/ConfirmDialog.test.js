import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ConfirmDialog from './ConfirmDialog';

describe('Component: ConfirmDialog', () => {
  let component = null;
  var place, open, onClose, onConfirm;

  beforeEach(() => {
    open = true;
    place = { name: 'test', open: '00:00', close: '23:59', description: '' };
    onClose = jest.fn();
    onConfirm = jest.fn();
  });

  const buildComponent = () => {
    const props = {
      place, open, onClose, onConfirm
    };

    return shallow(<ConfirmDialog {...props} />);
  };

  describe('Snapshots', () => {
    it('Should render correctly', () => {
      component = buildComponent();
      expect(shallowToJson(component)).toMatchSnapshot();
    });
  });

  describe('Actions', () => {
    it('Should trigger cancel when clicked the cancel button', () => {
      component = buildComponent();
      component.find('#confirm-cancel').simulate('click');
      expect(onClose).toHaveBeenCalledTimes(1);
    });

    it('Should trigger delete when clicked the delete button', () => {
      component = buildComponent();
      component.find('#confirm-delete').simulate('click');
      expect(onConfirm).toHaveBeenCalledTimes(1);
    });
  });
});