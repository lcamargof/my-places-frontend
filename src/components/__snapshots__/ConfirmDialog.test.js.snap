// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ConfirmDialog Snapshots Should render correctly 1`] = `
<WithStyles(Dialog)
  aria-describedby="alert-dialog-description"
  aria-labelledby="alert-dialog-title"
  onClose={[Function]}
  open={true}
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
     Remove a place :(
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)
      id="alert-dialog-description"
    >
      Are you sure you want to remove "
      test
      " from your list of places?
    </WithStyles(DialogContentText)>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      id="confirm-cancel"
      onClick={[Function]}
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      autoFocus={true}
      color="primary"
      id="confirm-delete"
      onClick={[Function]}
    >
      DELETE
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
